### PR TITLE
Fix internal temperature value incorrect

### PR DESCRIPTION
**Pull request by TDhaouST**
This ticket is to fix the incorrect temperature sensor (Die temp) value in the ADC device-tree node for the stm32 series below:
stm32U5
stm32L1
stm32L4
stm32L0
Acquired internal temperature values seems inconsistent and not compatible.

### CODE REVIEW

**Review by erwango — APPROVED** [review]

**Review by FRASTM — COMMENTED** [review]
with console output

File: dts/arm/st/u5/stm32u5.dtsi [hunk 1]
```diff
@@ -547,6 +547,7 @@
 			interrupts = <113 0>;
 			status = "disabled";
 			#io-channel-cells = <1>;
+			vref-mv = <3000>;
 			has-temp-channel;
```
Comment (FRASTM) [inline dts/arm/st/u5/stm32u5.dtsi:551]: The pb here is that TS CAL1 & TS CAL2  are calibration values acquired by ADC 1 with a VREF+ = 3V and the stm32U5 boards (nucleo as well as disco kit) have a VRef at 3.3V
Should it be ` ts-cal-vrefanalog = <3000>` and `vref-mv = <3000>;` ?
Again, the st,stm32-adc.yaml  defined the vref-mv as "the reference voltage of the ADC in mV (on the target board)." ==> better  set that value by the board DTS (not the SOC).

Comment (FRASTM) [reply dts/arm/st/u5/stm32u5.dtsi:551]: Second the ADC4 has 12 bits resolution and ADC1 has 14. The TS_CAL1 and TS_CAL2 are acquired with ADC1 on 14bits.

**Comment by FRASTM** [comment]
replaced by https://github.com/zephyrproject-rtos/zephyr/pull/50372

**Pull request closed without merging**